frontend/src/components/scheduleEntry/DialogActivityEdit.vue

```diff
@@ -44,46 +44,108 @@ export default {
+            periodOffset: scheduleEntry.periodOffset,
+            length: scheduleEntry.length,
+            key: scheduleEntry._meta.self,
+            deleted: false,
+            self: scheduleEntry._meta.self
+          }
+        }))
       }
     }
   },
   methods: {
+    /**
+     * Following code only works, if embedded collection patch is enabled & is working
+     * Blocked by https://github.com/api-platform/core/issues/4630
+     */
     /*
     updateActivity () {
-      return this.update()
-    },
-    update () {
-      this.error = null
-      const patchScheduleEntries = this.entityData.scheduleEntries
-        .map(entry => this.api.patch(entry._meta.self, {
+      const payloadData = {
+        ...this.entityData,
+
+        scheduleEntries: this.entityData.scheduleEntries?.map(entry => ({
+          period: entry.period()._meta.self,
           periodOffset: entry.periodOffset,
-          length: entry.length
-        }))
-      Promise.all(patchScheduleEntries)
-        .then(__ => ({ ...this.entityData }))
-        .then(entityData => {
-          delete entityData.scheduleEntries
-          return entityData
+          length: entry.length,
+          '@id': entry.self
+        })) || []
+      }
+
+      return this.update(payloadData)
+    }, */
+
+    updateActivity () {
+      this.error = null
+      const _events = this._events
+
+      const scheduleEntryPromises = this.entityData.scheduleEntries
+        .map(entry => {
+          // deleted local entry: do nothing
+          if (!entry.self && entry.deleted) {
+            return Promise.resolve()
+          }
+
+          // delete existing
+          if (entry.self && entry.deleted) {
+            return this.api.del(entry.self)
+          }
+
+          // update existing
+          if (entry.self) {
+            return this.api.patch(entry.self, {
+              period: entry.period()._meta.self,
+              periodOffset: entry.periodOffset,
+              length: entry.length
+            })
+          }
+
+          // else: create new entry
+          return this.scheduleEntries.$post({
+            period: entry.period()._meta.self,
+            periodOffset: entry.periodOffset,
+            length: entry.length,
+            activity: this.activity._meta.self
+          })
         })
-        .then(entityData => this.api.patch(this.entityUri, entityData))
-        .then(this.updatedSuccessful, this.onError)
+
+      // patch activity entity
+      const activityPayload = { ...this.entityData }
+      delete activityPayload.scheduleEntries
+
+      // first patch + post + delete all schedule entries
+      const promise = Promise.all(scheduleEntryPromises)
+
+        // then patch the activity itself (ensures after return we have a valid & complete activity in the local store)
+        .then(() => this.api.patch(this.entityUri, activityPayload))
+        .then(this.updatedSuccessful, e => this.onError(_events, e))
```

Comment:
im Fehlerfall könnte es sein, dass ein teil der scheduleEntries geupdated wurde.
Dann sollte man auch ein reload triggern.

Reply:
Ja, stimmt.
Lass uns heut diskutieren, in welche Richtung wir gehen wollen.

Reply:
Fixed with https://github.com/ecamp/ecamp3/pull/2393/commits/30b206adbc9412d95bba6caed48e7bd34d08b811